src/mango/src/mango_idx_view.erl

```diff
@@ -511,3 +512,44 @@ range_pos(Low, Arg, High) ->
                     max
             end
     end.
+
+
+maybe_filter_by_sort_fields(Idx, SortFields, Selector) ->
+    Cols = mango_idx:columns(Idx),
```

Comment:
I noticed this call was moved from ```mango_idx``` into each of three separate index types. You don't use ref ```Idx``` anywhere else, so perhaps you could move this line back and just pass in ```Cols``` as in a parameter? Having said that, in the future it might be useful if we need the entire index record for other purposes. If that is the case, then we can leave it here.

Reply:
I was in two minds here. Initially, I only passed in the `Cols` but then I thought it might be better to support Idx in case we need it for another reason. I don't really mind. Having Idx might future proof it a little.